staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -154,6 +155,43 @@ func allowsHTTP2(t *http.Transport) bool {
 	return false
 }
 
+// configureHttp2Transport explicitly configures a net/http HTTP/1 Transport
+// to use golang.org/x/net HTTP/2 instead of bundled HTTP/2.
+// It gives us an opportunity to configure the underlying HTTP/2 Transport.
+// It will return error if the HTTP/1 Transport has already been HTTP/2-enabled.
+func configureHttp2Transport(t1 *http.Transport) error {
+	h2t, err := http2.ConfigureTransports(t1)
+	if err != nil {
+		return err
+	}
+
+	// There are several confirmed flaws with Go's http2 implementation.
```

Comment:
I guess the flaws are in the specific networks. I agree golang's http2 keepalive implementation might have better defaults, but calling it a flaw seems rather harsh to me. "fatal situations" on the other hand hits the point. :+1: for references regarding why this is important!